test/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_service_test.clj

```diff
@@ -398,39 +398,50 @@
             :error))))))
 
 (deftest ^:integration debouncing-test
-  (testing "Default debouncing behavior"
+  (testing "debouncing"
     (let [root-dir (fs/temp-dir "debounce")
           actual-events (atom [])
-          actual-calls (atom 0)
+          callback-invocations (atom 0)
           callback (fn [e]
                      (swap! actual-events concat e)
-                     (swap! actual-calls inc))
+                     (swap! callback-invocations inc))
           app (tk/boot-services-with-config watch-service-and-deps {})
           svc (tk-app/get-service app :FilesystemWatchService)
-          event-cadence (quot watch-core/window-min 2)]
+          event-cadence (/ watch-core/window-min 2)]
       ;; We trigger half the number of events that could independently occur
       ;; without exceeding the window-max, given the fixed event-cadence
       ;; set above, we set the event-cadence to half of window-min to increase
       ;; the likelihood that they will be caught regardless of test runner
-      ;; speed. This requires that window-max be at least four times the
+      ;; speed.
+      ;;
+      ;; TODO what does "be caught" mean?
+      ;;
+      ;; This requires that window-max be at least four times the
       ;; duration of window-min.
-      (testing "multiple events trigger one callback when within `window-min` and `window-max`"
+      (testing "multiple events trigger one callback when within window-min and window-max"
         (let [test-dir (fs/file root-dir "window-min")
-              event-nums (quot (quot watch-core/window-max watch-core/window-min) 2)
-              files (map #(fs/file test-dir (str % ".txt")) (range 0 event-nums))
-              expected-events (set (map (fn [f] {:changed-path f :type :create}) files))]
-          (is (> event-nums 1))
+              ;; TODO shouldn't num-events be (quot (quot window-max event-cadence) 2) ?
+              ;; (based on the the previous comment)
+              num-events (quot (quot watch-core/window-max watch-core/window-min) 2)
+              files (for [n (range num-events)]
+                      (fs/file test-dir (str n ".txt")))
+              expected-events (set (for [f files] {:changed-path f
+                                                   :type :create}))]
+          ;; TODO this assertion strikes me as weird - it's the result of some
+          ;; division of constants - we should be able to say exactly what it is.
+          (is (> num-events 1))
```

Comment:
Not sure what you mean? If window-min is equal to or larger than 1/4 of window-max the num-events will be one or zero. Testing debouncing doesn't work well without multiple events to debounce.

I think there's some confusion because these tests are written to not care specifically about the window-min and window-max values. Using a cadence and numbers of events that are reasonable, I think, to assume, and failing the test if we change window-min and window-max in ways that mean those assumptions are no longer valid. However, that means we're computing a lot of values as the test runs and making assertions that our assumptions hold.

We could probably just hard code values for cadence and num-events that are reasonable for the known values of window-min and window-max and having a simple assertion that the values of window-min and window-max are what they currently are set to (with a note that if they change the values in this test might need to change to still be valid).